routing_deep_dive.prolific

```diff
@@ -0,0 +1,130 @@
+1. The Route Emitter on each Diego Cell polls Diego BBS for updated information about the apps on its Cell.
+1. ✨ The Route Emitter continually sends the route registration information it gets from the Diego BBS and the IP and port of the app to the GoRouter via NATS.
+1. ✨ The GoRouter keeps a mapping of routes -> ip:ports in a routes table, which is consulted when someone curls the route.
+
+### How?
+The following stories will look at how many components (CAPI, Diego BBS, Route Emitter, Nats, GoRouter) work together to make routes work.
+
+### Expected Result
+You can talk about route propagation at a high level.
+
+### Resources
+[CF Routes and Domains](https://docs.cloudfoundry.org/devguide/deploy-apps/routes-domains.html)
+
+
+L: routing
+---
+
+Subscribe to Route Emitter NATS Messages
+
+### What?
+There is one Route Emitter per Diego Cell and its job is to... emit routes.
+
+The Route Emitter tracks the desired route mappings for all of the apps it shares a Cell with. When a change is detected, the Route Emitter emits route registration and unregistration messages to the GoRouter via the NATS message bus. Even when no change is detected, the Route Emitter will periodically emit the current state of the world as a kind of heartbeat.
+
+For this story, let's look at the messages that the Route Emitter is publishing via NATS. Subscribing to these NATs messages can be a helpful debugging technique when it seems like routes are not being registered properly.
+
+### How?
+0. Bosh ssh onto a Diego Cell
+0. Download ruby and the NATS gem
+    ```
+    apt-get install -y ruby ruby-dev
+    gem install nats
+    ```
+0. Get the NATS username, password, and server address
+    ```
+    cat /var/vcap/jobs/route_emitter/config/route_emitter.json | jq . | grep nats
+    ```
+0. Use the nats gem to connect to nats:
+```nats-sub "*.*" -s nats://NATS_USERNAME:NATS_PASSWORD@NATS_ADDRESS```
+
+ The Route Emitter registers routes every 20 seconds (by default) so that the GoRouter (which subscribes to these messages) has the most up-to-date information about which IPs map to which apps and routes. Depending on how many routes there are, this might be a lot of information.
+
+0. Make a new route for one of your apps and find the route registration message.
+
+### Expected Result
+Inspect NATs messages. Look at what route information is sent to the GoRouter.
+
+## Resources
+[NATS message bus repo](https://github.com/nats-io/gnatsd)
+[NATS ruby gem repo](https://github.com/nats-io/ruby-nats)
+
+
+L: routing
+---
+
+Look at the GoRouter Routes Table
+
+### What?
+So the Route Emitter emits routes via the NATS message Bus. GoRouter _subscribes_ to those messages and keeps a route table that is uses to route network traffic bound for CF apps and CF components.
+
+Let's take a look at that route table.
+
+### How?
+0. Bosh ssh onto the router vm and become root.
+0. Install jq (a json manipulation and display tool)
+ ```
+ apt-get install jq
+ ```
```

Comment:
Indentation is a little inconsistent on the code blocks here as well. I duno how important it is in tracker markdown but it does effect github style markdown when considering where their indentation starts. (usually if they are actually nested under the bullet or not.)